pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -100,7 +98,7 @@ func (p *Provider) GetTemplate(ctx context.Context, nodeClass *v1alpha2.AKSNodeC
 	if err != nil {
 		return nil, err
 	}
-	launchTemplate, err := p.createLaunchTemplate(ctx, templateParameters)
+	launchTemplate, err := p.createLaunchTemplate(templateParameters)
```

Comment:
Is there a reason we dont want to store the context object on the Provider? I know go has this pattern of passing for each call so that you can cancel the context you created later, but we are really just using context for options and logging. 

Do we need to be passing it in the functions? I find myself adding and removing this frequently